symengine/sympy_compat.py

```diff
@@ -43,6 +47,35 @@ def __new__(cls, i):
         return symengine.Integer(i)
 
 
+class Float(Number):
+    _classes = (symengine.RealDouble, symengine.RealMPFR)
+
+    def __new__(cls, num, dps=None, prec=None, precision=None):
+        if dps is not None and precision is not None:
+            raise ValueError('Both decimal and binary precision supplied. '
+                             'Supply only one. ')
+        if dps is None and precision is None:
+            dps = 15
+            if isinstance(num, Float):
+                return num
```

Comment:
Yes, that would be consistent. Also, you can just remove the 2 lines above and the rest of the code below should handle that case.